### PR TITLE
Implement filter by field

### PR DESCRIPTION
**Pull request by vserret**

### CODE REVIEW

File: src/relation/builder.rs [hunk 1]
```diff
@@ -246,6 +246,47 @@ impl<RequireInput> MapBuilder<RequireInput> {
         builder
     }
 
+    /// Initialize a builder with an existing map and filter by an `Expr` that depends on the input columns
+    pub fn filter_field_with(self, map: Map, predicate: Expr) -> MapBuilder<WithInput> {
```
Comment (ngrislain) [inline src/relation/builder.rs:250]: Why is this called `filter_field`? you are filtering all fields right? It is a bit misleading.
Let's change this to `filter_with`. As a general naming principle, `filter` implicitly filter rows of data and `filter_fields` filter columns.

File: src/relation/builder.rs [hunk 2]
```diff
@@ -149,7 +149,7 @@ impl<RequireInput> MapBuilder<RequireInput> {
     }
 
     /// Initialize a builder with filtered existing map
-    pub fn filter_with<P: Fn(&str) -> bool>(self, map: Map, predicate: P) -> MapBuilder<WithInput> {
+    pub fn filter_fields_with<P: Fn(&str) -> bool>(self, map: Map, predicate: P) -> MapBuilder<WithInput> {
```
Comment (ngrislain) [inline src/relation/builder.rs:152]: yes

File: src/relation/builder.rs [hunk 3]
```diff
@@ -987,4 +1028,71 @@ mod tests {
             )
         }
     }
+
+    #[test]
+    fn test_map_filter_field_with() {
```
Comment (ngrislain) [inline src/relation/builder.rs:1033]: Rename

File: src/relation/transforms.rs [hunk 4]
```diff
@@ -703,6 +721,24 @@ impl Relation {
             .build()
     }
 
+    ///Returns a `Relation::Map` that inputs `self` and filter by `predicate`
+    pub fn filter_field(self, predicate: Expr) -> Relation {
```
Comment (vserret) [inline src/relation/transforms.rs:725]: make it private

File: src/relation/transforms.rs [hunk 5]
```diff
@@ -71,17 +73,33 @@ impl Map {
...
     /// Map fields
     pub fn map_fields<F: Fn(&str, Expr) -> Expr>(self, f: F) -> Map {
         Relation::map().map_with(self, f).build()
     }
-
     /// Rename fields
     pub fn rename_fields<F: Fn(&str, Expr) -> String>(self, f: F) -> Map {
         Relation::map().rename_with(self, f).build()
     }
+    /// Returns a `Map` that inputs `Relation::Map(self)` and filter by `predicate`
```
Comment (vserret) [inline src/relation/transforms.rs:86]: filter

**Pull request merged**